src/jupyterlabcontroller/storage/gafaelfawr.py

```diff
@@ -0,0 +1,50 @@
+from typing import Any, List, Optional
+
+from httpx import AsyncClient
+
+from ..models.v1.lab import UserInfo
+
+
+class GafaelfawrStorageClient:
+    def __init__(self, token: str, http_client: AsyncClient) -> None:
+        self.token = ""
+        self.http_client = http_client
+        self._user: Optional[UserInfo] = None
+        self._api_url = "/auth/api/v1"
+        self.set_token(token)
```

Comment:
`token` shouldn't be a constructor variable. This is creating a per-user storage object, but that's more complexity than you need. A single `GafaelfawrStorageClient` can look up any user. If you make `get_user` and `get_scopes` take the token as an argument, you can delete `_reset_token_info`, `set_token`, and all of that logic and make the class much simpler. (`get_token` should also go; the caller already has the token by definition, so it shouldn't need to ask the class for it).

Reply:
Done